compiler-plugin/src/main/java/io/ballerina/wso2/apim/catalog/codemodifier/OpenAPIAnnotationModifier.java

```diff
@@ -0,0 +1,249 @@
+            NodeList<AnnotationNode> annotations = getMetadataNode(serviceNode).annotations();
+            for (AnnotationNode annotation: annotations) {
+                if (isServiceCatalogConfigAnnotationAvailable(annotation)) {
+                    return imports;
+                }
+            }
+        }
+        if (containsServiceNode) {
+            return imports.add(createImportDeclarationNodeForModule());
+        }
+        return imports;
+    }
+
+    private NodeList<ModuleMemberDeclarationNode> updateMemberNodes(
+            SyntaxTree syntaxTree, Package currentPackage, Document document, Project project,
+            NodeList<ModuleMemberDeclarationNode> oldMembers, SemanticModel semanticModel) {
+        List<ModuleMemberDeclarationNode> updatedMembers = new LinkedList<>();
+        for (ModuleMemberDeclarationNode memberNode : oldMembers) {
+            if (memberNode.kind() != SyntaxKind.SERVICE_DECLARATION) {
+                updatedMembers.add(memberNode);
+                continue;
+            }
+
+            ServiceDeclarationNode serviceNode = (ServiceDeclarationNode) memberNode;
+            String openAPIDef = generateOASForgetServiceDeclarationNode(
+                    serviceNode, syntaxTree, semanticModel, project, currentPackage, document);
+            MetadataNode metadataNode = getMetadataNode(serviceNode);
+            MetadataNode.MetadataNodeModifier modifier = metadataNode.modify();
+            NodeList<AnnotationNode> updatedAnnotations = updateAnnotations(metadataNode.annotations(),
+                    openAPIDef);
+            modifier.withAnnotations(updatedAnnotations);
+            MetadataNode updatedMetadataNode = modifier.apply();
+            ServiceDeclarationNode.ServiceDeclarationNodeModifier serviceDecModifier = serviceNode.modify();
+            serviceDecModifier.withMetadata(updatedMetadataNode);
+            ServiceDeclarationNode updatedServiceDecNode = serviceDecModifier.apply();
+            updatedMembers.add(updatedServiceDecNode);
+        }
+        return AbstractNodeFactory.createNodeList(updatedMembers);
+    }
+
+    private NodeList<AnnotationNode> updateAnnotations(NodeList<AnnotationNode> currentAnnotations,
+                                                       String openApiDefinition) {
+        NodeList<AnnotationNode> updatedAnnotations = NodeFactory.createNodeList();
+        boolean serviceCatalogConfigAnnotationUpdated = false;
+        for (AnnotationNode annotation: currentAnnotations) {
+            if (isServiceCatalogConfigAnnotationAvailable(annotation)) {
+                serviceCatalogConfigAnnotationUpdated = true;
+                SeparatedNodeList<MappingFieldNode> updatedFields = getUpdatedFields(annotation, openApiDefinition);
+                MappingConstructorExpressionNode annotationValue =
+                        annotation.annotValue().get().modify().withFields(updatedFields).apply();
```

Comment:
`annotation.annotValue()` is Optional right? we have to check and ifPresent() before get() right?

Reply:
`if (isServiceCatalogConfigAnnotationAvailable(annotation)) {` when this condition true, it will always present ryt?